libs/widget-lib/src/IframeCowEventEmitter.ts

```diff
@@ -0,0 +1,42 @@
+import { CowEventEmitterImpl, CowEventListener, CowEventListeners, CowEvents } from '@cowprotocol/events'
+const COW_SWAP_WIDGET_EVENT_KEY = 'cowSwapWidget'
+
+export class IframeCowEventEmitter {
+  private listeners: CowEventListeners
+  private eventEmitter: CowEventEmitterImpl
+
+  constructor(listeners?: CowEventListeners) {
+    this.eventEmitter = new CowEventEmitterImpl()
+    this.listeners = listeners ?? []
```

Comment:
Nitpick: do we ever need to keep the original passed value if `''` or `0`?

```suggestion
    this.listeners = listeners || []
```

Reply:
listener is an array, it doesn't make any difference one or the other. Anyways, changing it

Reply:
9274bbd27083ad690860621759329a491f68c9f5

Reply:
9274bbd27083ad690860621759329a491f68c9f5

Reply:
You can just define it as:
```
private listeners: CowEventListeners = []
```